storybook/src/components/TextInput/TextInput.docs.mdx

```diff
@@ -14,14 +14,116 @@ import README from "../../../../packages/css/src/components/text-input/README.md
 
 ## Examples
 
+### Type: Password
```

Comment:
I'm not sure we should allow `type="password"` yet. Maybe we should decide first if we want a password input with show/hide text?

Reply:
Why wouldn't we allow `type="password"` yet?

Reply:
A password field requires quite a bit of configuration, maybe enough to warrant a separate component? Especially if we add a show/hide button.

But let's go ahead with this PR now, and discuss this point before the next release.